src/ext/sse.js

```diff
@@ -170,23 +178,28 @@ This extension adds support for Server Sent Events to htmx.  See /www/extensions
 				var sseEventNames = getLegacySSESwaps(child);
 			}
 
-			for (var i = 0 ; i < sseEventNames.length ; i++) {
+			for (var i = 0; i < sseEventNames.length; i++) {
 				var sseEventName = sseEventNames[i].trim();
 				var listener = function(event) {
 
-					// If the parent is missing then close SSE and remove listener
-					if (maybeCloseSSESource(elt)) {
-						source.removeEventListener(sseEventName, listener);
+					// If the source is missing then close SSE
+					if (maybeCloseSSESource(sourceElement)) {
```

Comment:
The approach with `sse-swap` here is the correct one (using `queryAttributeOnThisOrChildren`), and it should be applied to `sse-connect` as well. Just make sure to perform a separate `getClosestMatch` lookup for SSE event source instead of reusing `sourceElement` in this line and the whole relevant block of code